File: medplum-java/medplum-model/src/main/java/com/fhir/SpecimenDefinition_Handling.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSpecimenDefinition_Handling.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SpecimenDefinition_Handling {
  Optional<String> instruction();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  Optional<Range> temperatureRange();

  Optional<String> id();

  Optional<Duration> maxDuration();

  Optional<CodeableConcept> temperatureQualifier();

  static ImmutableSpecimenDefinition_Handling.Builder builder() {
    return ImmutableSpecimenDefinition_Handling.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/SpecimenDefinition_Handling.ts
import { CodeableConcept } from "./CodeableConcept";
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { Range } from "./Range";

export interface SpecimenDefinition_Handling {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    temperatureQualifier: CodeableConcept | null | undefined;
    temperatureRange: Range | null | undefined;
    maxDuration: Duration | null | undefined;
    instruction: string | null | undefined;
}
